rpm/stackdriver-agent

```diff
@@ -118,57 +164,107 @@ start () {
     fi
 
     if ! check_config; then
-        echo >&2 "Not starting, configuration/credentials error."
-        return 1
+        log_failure_msg "not starting, configuration/credentials error."
+        return 3
```

Comment:
Should we return `2 if the daemon could not be started` instead of `3 if the daemon was not supposed to be started` here? `3` sounds like things are working as expected, while we are in fact failing the start, and something needs to be fixed from user side.

Reply:
Hmm, below we treat 3 as a (different kind of) failure. The code currently returns 3 if the daemon doesn't start because of user-supplied configuration (as opposed to some other problem, e.g., corruption or permission error, running out of memory, etc).

Reply:
Cool.

[Optional] We could clarify the comment in line 153. Something like `3 if the daemon was not supposed to be started or doesn't start because of user-supplied configuration`.

Reply:
Good idea. Added a variant of that to both scripts.